paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc

```diff
@@ -308,7 +326,7 @@ void ExecuteFusedMatmul(const OneDNNContext &dev_ctx,
         residual_data_vec[3] > 1) {
       residual_data_memory_p = handler.AcquireSrcMemoryStride(residual_data);
     } else {
-      residual_data_memory_p = handler.AcquireSrcMemory(residual_data);
+      residual_data_memory_p = handler.AcquireSrcMemoryResidual(residual_data);
```

Comment:
Assume the formula is src * weight + residual, AcquireSrcMemory will use src mem_desc to create memory , not residual mem_desc. I think it's a bug in used implement, since residual dims can different with src dims.

Reply:
do we have a general func to acquire mdesc. It seems the newly added func is not limited with residual memory. It create memory based on the input of the func.

Reply:
Well, seems don't have such general func now, you can see that in onednn_reuse.h, all mdesc is confirmed in advance.
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/phi/backends/onednn/onednn_reuse.h#L200
I found in conv kernel,  we also have to create such func to acquire residual memory.

Reply:
okay. got u. I'm not sure if the residual memory desc should be also queried from the pd like what src does and then reorder if the layout is not the same. Can you please double comfirm on it? cc @YangQun1

Reply:
In my investigation, it seems that no pd in oneDNN can support post-op desc acquisition like src_desc() or weights_desc()? 
So do we also need to reorder manually for dst and residual in this issue?